### PR TITLE
add dependency for markdown service.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @superyyrrzz @hellosnow @ansyral @qinezh 
#449

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/HostService.cs [hunk 1]
```diff
@@ -178,6 +187,78 @@ where string.Equals(attr.Name, "href", StringComparison.OrdinalIgnoreCase) || st
...
+                                 from attr in n.Attributes
+                                 where string.Equals(attr.Name, "href", StringComparison.OrdinalIgnoreCase) || string.Equals(attr.Name, "uid", StringComparison.OrdinalIgnoreCase)
+                                 where !string.IsNullOrWhiteSpace(attr.Value)
+                                 select attr.Value).ToImmutableHashSet();
+            using (var sw = new StringWriter())
+            {
+                doc.Save(sw);
+                result.Html = sw.ToString();
+            }
+            return result;
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/HostService.cs]: extract to a method?

Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/HostService.cs]: call ParseHtml(markupResult.Html)?

Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/HostService.cs]: oh, are you gonna remove this method from the interface in the future?

**Comment by ansyral** [comment]
:+1:

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by qinezh** [comment]
:shipit:

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Engine/HostService.cs (hunk 1 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/HostService.cs]: yes, we need dependency information.

**Pull request merged**